### PR TITLE
Feature/sfw 367

### PR DESCRIPTION
**Pull request by lucaangeletti**
# Changes

- Fix to make `BasePresenter` compatible with the weak `view` property
- Documentation for: `BasePresenter`, `HasPresenter`, `MetaModule`,  `Wireframe`

[Jira ticket](https://stanwood.atlassian.net/browse/SFW-367)

### CODE REVIEW

File: StanwoodCore/Protocols/Wireframe.swift [hunk 1]
```diff
@@ -25,16 +25,28 @@
...
 
-/// :nodoc:
-/*public*/ class Wireframe<M: MetaModule> {
 
-    /// :nodoc:
-    /*public*/ static func makeViewController() -> M.ViewController {
+/// This generic class provide a way of instantiating all the objects needed for a given module.
+///
+/// ```
+/// // Example:
```
Comment (talezion) [inline StanwoodCore/Protocols/Wireframe.swift]: @lucaangeletti Let's use the correct format for `Example`

https://developer.apple.com/library/archive/documentation/Xcode/Reference/xcode_markup_formatting_ref/Example.html#//apple_ref/doc/uid/TP40016497-CH58-SW1

File: StanwoodCore/Protocols/BasePresenter.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+import Foundation
+
+
+/// BasePresenter offers the minimum properties (actions, parameters, view) required for a type conforming to Presentable.
+/// The developer should create its own ModulePresenter type simply sublassing BasePresenter. This will avoid lots of boilerplate code and leave ModulePresenter free to contains only the code relevant to the current module.
+class BasePresenter<A, P, V: AnyObject>: Presentable {
```
Comment (talezion) [inline StanwoodCore/Protocols/BasePresenter.swift]:
```suggestion
public class BasePresenter<A, P, V: AnyObject>: Presentable {
```

File: StanwoodCore/Protocols/HasPresenter.swift [hunk 3]
```diff
@@ -25,8 +25,12 @@
...
-/// :nodoc:
-/*public*/ protocol HasPresenter: AnyObject {
+/// A protocol having a *presenter* property of type Presenter.
+/// Specifically this protocol is used to define at compile time an **has** relationship between a UIViewController and a Presenter
+public protocol HasPresenter: AnyObject {
+  
     associatedtype Presenter
+  
+    /// A read/write property of type Presenter
     var presenter: Presenter! { get set }
```
Comment (talezion) [inline StanwoodCore/Protocols/HasPresenter.swift]: @lucaangeletti do we need this public?

Comment (lucaangeletti) [reply StanwoodCore/Protocols/HasPresenter.swift]: @talezion I think we can keep this property `internal`.

Comment (lucaangeletti) [reply StanwoodCore/Protocols/HasPresenter.swift]: @talezion Let's leave the entire protocol `public` 👍

File: StanwoodCore/Protocols/Wireframe.swift (hunk 1 above)
Comment (lucaangeletti) [reply StanwoodCore/Protocols/Wireframe.swift]: @talezion Unfortunately that markup still does not work with Xcode, it's Playground only

**Pull request merged**